apps/desktop/src/main/window.main.ts

```diff
@@ -211,21 +246,42 @@ export class WindowMain {
     }
 
     // Show it later since it might need to be maximized.
-    this.win.show();
-
-    // and load the index.html of the app.
-    // FIXME: Verify that this floating promise is intentional. If it is, add an explanatory comment and ensure there is proper error handling.
-    // eslint-disable-next-line @typescript-eslint/no-floating-promises
-    this.win.loadURL(
-      url.format({
-        protocol: "file:",
-        pathname: path.join(__dirname, "/index.html"),
-        slashes: true,
-      }),
-      {
-        userAgent: cleanUserAgent(this.win.webContents.userAgent),
-      },
-    );
+    // use once event to avoid flash on unstyled content.
+    this.win.once("ready-to-show", () => {
+      this.win.show();
+    });
+
+    if (template === "full-app") {
+      // and load the index.html of the app.
+      // FIXME: Verify that this floating promise is intentional. If it is, add an explanatory comment and ensure there is proper error handling.
+      // eslint-disable-next-line @typescript-eslint/no-floating-promises
+      this.win.loadURL(
```

Comment:
suggestion: fix the FIXME while you're at it
```suggestion
      // FIXME: Verify that this floating promise is intentional. If it is, add an explanatory comment and ensure there is proper error handling.
      // eslint-disable-next-line @typescript-eslint/no-floating-promises
      void this.win.loadURL(
```

Reply:
Or actually, maybe add `await` since we're awaiting in the `else` block

Reply:
The reason I didn't touch this was because of that comment - I didn't want to touch something I had no experience with and was unrelated to my experiment in case it would have bad side effects.

Reply:
I don't think anybody owns this file so let's just improve it while we're at it :) It's a great opportunity since we're gonna get lots of QA on this anyways. I'd suggest adding `void` to keep the existing behavior (I'm assuming you haven't encountered any issues with it)

Reply:
Done - I also removed the eslint disable.